### ISSUE TITLE
Method "__eq__" and "__ne__" overrides class "object" in an incompatible manner regression between 1.1.160 and 1.1.161

### ISSUE THREAD

**Issue by bloussou** [microsoft/pyright#2287]
In 1.1.160 the below code was not raising any issue. 
In 1.1.161 and above I get : 
```python
error: Method "__eq__" overrides class "object" in an incompatible manner
    Return type mismatch: base method returns type "bool", override returns type "str"
      "str" is incompatible with "bool" (reportIncompatibleMethodOverride)
error: Method "__ne__" overrides class "object" in an incompatible manner
    Return type mismatch: base method returns type "bool", override returns type "str"
      "str" is incompatible with "bool" (reportIncompatibleMethodOverride)
```

```python
from typing import Any

class A:

    def __eq__(self, other: Any) -> Any:
        pass

    def __ne__(self, other: Any) -> Any:
        pass

class B(A):

    def __eq__(self, other: Any) -> str:
        return ""

    def __ne__(self, other: Any) -> str:
        return ""
```

**VS Code extension or command-line**
I'm using the CLI

**Comment by erictraut** [comment]
I think the current behavior is correct. You're overriding the `__eq__` and `__ne__` methods in the `object` class in an incompatible manner. Mypy emits the same error in this case.

If you want, you can disable this diagnostic check locally to the file using `# pyright: reportIncompatibleMethodOverride=false`.

**Comment by bloussou** [comment]
Ok got it, thanks for your help !

**Comment by bloussou** [comment]
@erictraut are you sure about this, according to the python doc https://docs.python.org/3/reference/datamodel.html?highlight=__eq__#object.__eq__ `By convention, False and True are returned for a successful comparison. However, these methods can return any value, so if the comparison operator is used in a Boolean context (e.g., in the condition of an if statement), Python will call bool() on the value to determine if the result is true or false.`. 
I cannot reopen this issue

**Comment by bloussou** [comment]
Pandas is returning `Series` for instance
![image](https://user-images.githubusercontent.com/32448554/133261047-2b2c0fd3-5e0b-42a6-b0f0-0a58506f58a6.png)

**Comment by erictraut** [comment]
From the perspective of the `reportIncompatibleMethodOverride` diagnostic check, pyright is doing the right thing here. You are attempting to override an existing method with an incompatible signature, and pyright is correctly detecting that and reporting it to you. Mypy reports the same error in this case.

If you believe that the `__eq__` and `__ne__` methods defined in the `object` class are annotated incorrectly and should return a type other than `bool`, please file a bug in the typeshed repo.
